compiler/AST/FnSymbol.cpp

```diff
@@ -780,6 +801,10 @@ bool FnSymbol::hasGenericFormals(SymbolMap* map) const {
     if (formal->intent == INTENT_PARAM) {
       isGeneric = true;
 
+    } else if (isConstrainedType(formal->type)) {
+      if (isConstrainedGeneric())
+        isGeneric = true;
```

Comment:
Should this have an `else` clause to do an `INT_FATAL` ? Is it possible to have a formal type that is a constrained type outside of a constrained generic?